File: applications/sinks/event-account-jdbc-sink/src/test/java/com/vmware/account/jdbc/sink/consumer/AccountConsumerTest.java
package com.vmware.account.jdbc.sink.consumer;

import com.vmware.account.jdbc.sink.domain.AccountEntity;
import com.vmware.account.jdbc.sink.repository.AccountRepository;
import nyla.solutions.core.patterns.creational.generator.JavaBeanGeneratorCreator;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.mockito.Mock;
import org.mockito.junit.jupiter.MockitoExtension;

import static org.mockito.ArgumentMatchers.any;
import static org.mockito.Mockito.verify;

@ExtendWith(MockitoExtension.class)
class AccountConsumerTest {

    private AccountConsumer subject;
    @Mock
    private AccountRepository repository;
    private AccountEntity account = JavaBeanGeneratorCreator.of(AccountEntity.class).create();

    @BeforeEach
    void setUp() {
        subject = new AccountConsumer(repository);
    }

    @Test
    void accept() {
        subject.accept(account);

        verify(repository).save(any());
    }
}

File: applications/account-geode-sink/src/test/kotlin/com/vmware/financial/open/banking/account/consumer/AccountConsumerTest.kt
package com.vmware.financial.open.banking.account.consumer

import com.vmware.financial.open.banking.account.domain.Account
import com.vmware.financial.open.banking.account.service.AccountService
import nyla.solutions.core.patterns.creational.generator.JavaBeanGeneratorCreator
import org.junit.jupiter.api.Test

import org.junit.jupiter.api.BeforeEach
import org.mockito.kotlin.any
import org.mockito.kotlin.mock
import org.mockito.kotlin.verify

internal class AccountConsumerTest {

    private lateinit var accountService: AccountService
    private lateinit  var account : Account

    @BeforeEach
    internal fun setUp() {
        accountService = mock<AccountService>()
        account = JavaBeanGeneratorCreator.of(Account::class.java).create()
    }

    @Test
    fun accept() {
        account = JavaBeanGeneratorCreator.of(Account::class.java).create()
        var subject = AccountConsumer(accountService)
        subject.accept(account)

        verify(accountService).createAccount(any<Account>())
    }
}